framework/web/Controller.php

```diff
@@ -162,6 +163,19 @@ public function bindActionParams($action, $params)
                 }
                 $args[] = $actionParams[$name] = $params[$name];
                 unset($params[$name]);
+            } elseif (PHP_VERSION_ID >= 70100 && ($type = $param->getType()) !== null && !$type->isBuiltin()) {
```

Comment:
Can't it work with lower PHP versions?

Reply:
It probably could, but honestly I don't want to spend time writing code for ancient runtime versions. These platforms are likely in a maintenance phase and have no need for the latest features.

Reply:
Isn't PHP 5 exactly the same except not having scalar type-hints, in this regard?

Reply:
Nullable types are different as well as the reflection API.
Instead of making the code more complex I chose to keep it simple and focusing in active developers instead. Platforms on PHP5 are likely not seeing active development and don't need to adopt this new functionality.